### PR TITLE
retry updating state if container dies during an outage

### PR DESCRIPTION
**Pull request by smousa**

### CODE REVIEW

File: zzk/service/state.go [hunk 1]
```diff
@@ -560,8 +560,12 @@ func DeleteState(conn client.Connection, req StateRequest) error {
 	// Delete the host instance
 	hspth := path.Join(basepth, "/hosts", req.HostID, "instances", req.StateID())
 	if ok, err := conn.Exists(hspth); err != nil {
-
 		logger.WithError(err).Debug("Could not look up host state")
+
+		// CC-2853: only wrap errors that are of type client.ErrNoServer
```
Comment (johnlhamilton) [inline zzk/service/state.go]: Should this be "are NOT of type client.ErrNoServer"?

File: zzk/service/state.go [hunk 2]
```diff
@@ -576,8 +580,12 @@ func DeleteState(conn client.Connection, req StateRequest) error {
 	// Delete the service instance
 	sspth := path.Join(basepth, "/services", req.ServiceID, req.StateID())
 	if ok, err := conn.Exists(sspth); err != nil {
-
 		logger.WithError(err).Debug("Could not look up service state")
+
+		// CC-2853: only wrap errors that are of type client.ErrNoServer
+		if err == client.ErrNoServer {
```
Comment (johnlhamilton) [inline zzk/service/state.go]: Same

File: zzk/service/state.go [hunk 3]
```diff
@@ -590,8 +598,12 @@ func DeleteState(conn client.Connection, req StateRequest) error {
 	}
 
 	if err := t.Commit(); err != nil {
-
 		logger.WithError(err).Debug("Could not commit transaction")
+
+		// CC-2853: only wrap errors that are of type client.ErrNoServer
```
Comment (johnlhamilton) [inline zzk/service/state.go]: Ditto

Comment (smousa) [inline zzk/service/state.go]: Yes, you are right

**Pull request merged**